### PR TITLE
Add assetUrl to track data

### PR DESCRIPTION
**Pull request by taylorkearns**
Adds the assetUrl (`ipod-library://item/item.mp3?id=12345`) to the track objects returned from `getTracks`.

### CODE REVIEW

**Comment by taylorkearns** [comment]
@n1ru4l would love any feedback from you as well if you are interested. I am not very versed in obj-c.

File: RNiTunes/RNiTunes.m [hunk 1]
```diff
@@ -196,7 +205,15 @@ @implementation RNiTunes
         NSDictionary *songDictionary = [NSMutableDictionary dictionary];
 
         if (fields == nil) {
-            songDictionary = @{@"albumTitle":albumTitle, @"albumArtist": albumArtist, @"duration":[duration isKindOfClass:[NSString class]] ? [NSNumber numberWithInt:[duration intValue]] : duration, @"genre":genre, @"playCount": [NSNumber numberWithInt:[playCount intValue]], @"title": title};
+            songDictionary = @{
+              @"albumTitle": albumTitle,
+              @"albumArtist": albumArtist,
+              @"assetUrl": assetUrl,
```
Comment (kevin-j-m) [inline RNiTunes/RNiTunes.m:211]: This is the only difference from the prior line, correct? Just want to make sure I'm not missing anything else.

File: RNiTunes/RNiTunes.m [hunk 2]
```diff
@@ -149,6 +149,10 @@ @implementation RNiTunes
         NSString *searchGenre = [query objectForKey:@"genre"];
         [songsQuery addFilterPredicate:[MPMediaPropertyPredicate predicateWithValue:searchGenre forProperty:MPMediaItemPropertyGenre comparisonType:MPMediaPredicateComparisonContains]];
     }
+    if ([query objectForKey:@"assetUrl"] != nil) {
```
Comment (kevin-j-m) [inline RNiTunes/RNiTunes.m:152]: What does `@` do?

File: RNiTunes/RNiTunes.m [hunk 3]
```diff
@@ -149,6 +149,10 @@ @implementation RNiTunes
...
+        NSString *searchAssetUrl = [query objectForKey:@"assetUrl"];
+        [songsQuery addFilterPredicate:[MPMediaPropertyPredicate predicateWithValue:searchAssetUrl forProperty:MPMediaItemPropertyAssetURL comparisonType:MPMediaPredicateComparisonContains]];
```
Comment (kevin-j-m) [inline RNiTunes/RNiTunes.m:154]: I'm guessing this is building up the `song` object that's referenced below, is that correct?

File: RNiTunes/RNiTunes.m [hunk 4]
```diff
@@ -166,6 +170,8 @@ @implementation RNiTunes
         NSString *genre = [song valueForProperty: MPMediaItemPropertyGenre]; // filterable
         NSString *duration = [song valueForProperty: MPMediaItemPropertyPlaybackDuration];
         NSString *playCount = [song valueForProperty: MPMediaItemPropertyPlayCount];
+        NSURL *url = [song valueForProperty: MPMediaItemPropertyAssetURL];
+        NSString *assetUrl = url.absoluteString;
```
Comment (kevin-j-m) [inline RNiTunes/RNiTunes.m:174]: What's the value building up all these properties on this one object and then pretty shortly dereferencing the same properties to construct these different strings?

**Review by kevin-j-m — COMMENTED** [review]
This is the first Obj-C I've read, so most of this is just for my own understanding. However, unfortunately, due to that, I don't feel comfortable commenting on the correctness or degree to which this succeeds in its goal.

File: RNiTunes/RNiTunes.m [hunk 5]
```diff
@@ -196,7 +205,15 @@ @implementation RNiTunes
...
+            songDictionary = @{
+              @"albumTitle": albumTitle,
+              @"albumArtist": albumArtist,
+              @"assetUrl": assetUrl,
+              @"duration": [duration isKindOfClass:[NSString class]] ? [NSNumber numberWithInt:[duration intValue]] : duration,
+              @"genre": genre,
+              @"playCount": [NSNumber numberWithInt:[playCount intValue]],
+              @"title": title
+            };
         } else {
```
Comment (kevin-j-m) [inline RNiTunes/RNiTunes.m:217]: Do you need to do anything with `assertUrl` in this case?

File: RNiTunes/RNiTunes.m (hunk 2 above)
Comment (n1ru4l) [reply RNiTunes/RNiTunes.m:152]: Initiating a constant string declaration

File: RNiTunes/RNiTunes.m (hunk 1 above)
Comment (taylorkearns) [reply RNiTunes/RNiTunes.m:211]: Yes, thanks for pointing out. The only code change is this addition. Rest is formatting.

File: RNiTunes/RNiTunes.m [hunk 6]
```diff
@@ -196,7 +205,15 @@ @implementation RNiTunes
         NSDictionary *songDictionary = [NSMutableDictionary dictionary];
 
         if (fields == nil) {
-            songDictionary = @{@"albumTitle":albumTitle, @"albumArtist": albumArtist, @"duration":[duration isKindOfClass:[NSString class]] ? [NSNumber numberWithInt:[duration intValue]] : duration, @"genre":genre, @"playCount": [NSNumber numberWithInt:[playCount intValue]], @"title": title};
+            songDictionary = @{
+              @"albumTitle": albumTitle,
+              @"albumArtist": albumArtist,
+              @"assetUrl": assetUrl,
+              @"duration": [duration isKindOfClass:[NSString class]] ? [NSNumber numberWithInt:[duration intValue]] : duration,
```
Comment (jyurek) [inline RNiTunes/RNiTunes.m:212]: Don't you know what the duration is, though? You define/load it above. Edit: Oh this isn't your code.

File: RNiTunes/RNiTunes.m (hunk 3 above)
Comment (jyurek) [reply RNiTunes/RNiTunes.m:154]: That's a good question. I would assume that, too, but I don't see any use of `songsQuery` after this.

Comment (taylorkearns) [reply RNiTunes/RNiTunes.m:154]: I believe this is just building up the [songDictionary](https://github.com/kadiks/react-native-itunes/blob/087941ce55633baeb54bb33d4e77923715270137/RNiTunes/RNiTunes.m#L208) either using the `query` param or the `fields` param.

File: RNiTunes/RNiTunes.m (hunk 4 above)
Comment (taylorkearns) [reply RNiTunes/RNiTunes.m:174]: Not sure, maybe just so we can reference the vars and put them up here?

File: RNiTunes/RNiTunes.m (hunk 5 above)
Comment (taylorkearns) [reply RNiTunes/RNiTunes.m:217]: Yea, way down [here](https://github.com/kadiks/react-native-itunes/blob/087941ce55633baeb54bb33d4e77923715270137/RNiTunes/RNiTunes.m#L386).

**Pull request closed without merging**